src/demandlib/bdew/elec_slp.py

```diff
@@ -109,19 +109,20 @@ def create_bdew_load_profiles(self, dt_index, slp_types, holidays=None):
         tmp_df.set_index(index, inplace=True)
 
         # Create empty DataFrame to take the results.
-        new_df = pd.DataFrame(index=dt_index, columns=slp_types).fillna(0)
+        new_df = pd.DataFrame(index=dt_index, columns=slp_types, dtype=float).fillna(0)
         new_df = add_weekdays2df(
             new_df, holidays=holidays, holiday_is_sunday=True
         )
 
-        new_df["hour"] = dt_index.hour
-        new_df["minute"] = dt_index.minute
+        new_df["hour"] = dt_index.hour.astype(float)
+        new_df["weekday"] = new_df["weekday"].astype(int)
+        new_df["minute"] = dt_index.minute.astype(float)
         time_df = new_df[["date", "hour", "minute", "weekday"]].copy()
         tmp_df[slp_types] = tmp_df[slp_types].astype(float)
 
         # Inner join the slps on the time_df to the slp's for a whole year
-        tmp_df["hour_of_day"] = tmp_df.index.hour
-        tmp_df["minute_of_hour"] = tmp_df.index.minute
+        tmp_df["hour_of_day"] = tmp_df.index.hour.astype(float)
+        tmp_df["minute_of_hour"] = tmp_df.index.minute.astype(float)
```

Comment:
Shouldn't it make sense to preserve the integers, here?

```suggestion
        tmp_df["hour_of_day"] = tmp_df.index.hour.astype(int)
        tmp_df["minute_of_hour"] = tmp_df.index.minute.astype(int)
```